spec/services/events/create_service_spec.rb

```diff
@@ -116,5 +116,22 @@
         expect(result.event.timestamp.iso8601(3)).to eq('2023-09-04T15:45:12.344Z')
       end
     end
+
+    context 'when kafka is configured' do
+      let(:karafka_producer) { instance_double(WaterDrop::Producer) }
```

Comment:
You can test this integration with a buffered client.

Basically you can overwrite partially waterdrop setup in karafka:

https://karafka.io/docs/WaterDrop-reconfiguration/#partial-reconfiguration

and then you can in your specs have a producer that buffers messages. That way you can check if they are dispatches where expected, etc.

```ruby
WaterDrop::Producer.new do |config|
  # This will make it buffer instead of sending to kafka
  config.client_class = WaterDrop::Clients::Buffered
end

# it tests using buffered client
Karafka.producer.produce_async(topic: 'x', payload: 'b')

expect(Karafka.producer.client.messages.last[:topic]).to eq('x') # etc
```

Reply:
Sure. Feel free to tag mi in the PRs if you need any help :) happy to provide some insights as karafka can be sometimes complex.